lib/ConditionBuilder.js

```diff
@@ -19,14 +28,138 @@ ConditionBuilder.populateFieldFromConditionBuilderList = function (field, condit
     assert.ok(Array.isArray(conditions), 'Expected array')
     for (var i = 0; i < conditions.length; i++) {
       assert.ok(conditions[i] instanceof ConditionBuilder, 'Expected ConditionBuilder')
-      for (var key in conditions[i]._conditions) {
-        field[key] = conditions[i]._conditions[key]
+      for (var attr in conditions[i]._conditions) {
+        field[attr] = conditions[i]._conditions[attr]
       }
     }
   }
 }
 
 /**
+ * @param {Object} data
+ * @param {string} fieldName The fieldName on `data`
+ * @param {Array<ConditionBuilder>} conditions
+ * @param {{count: number}} nameMutex
+ * @return {ConditionBuilder|ConditionJunction}
+ */
+ConditionBuilder.populateExpressionField = function (data, fieldName, conditions, nameMutex) {
+  ConditionBuilder.validateConditions(conditions)
+
+  var junction = new ConditionJunction('AND', conditions)
+  junction.assignUniqueNames(nameMutex)
+
+  if (!data.ExpressionAttributeNames) {
+    data.ExpressionAttributeNames = {}
+  }
+  typeUtil.extendAttributeNames(data.ExpressionAttributeNames, junction.buildAttributeNames())
+
+  if (!data.ExpressionAttributeValues) {
+    data.ExpressionAttributeValues = {}
+  }
+  typeUtil.extendAttributeValues(data.ExpressionAttributeValues, junction.buildAttributeValues())
+
+  data[fieldName] = junction.buildExpression()
+  return junction
+}
+
+/**
+ * @param {Object} nameMutex
+ */
+ConditionBuilder.prototype.assignUniqueNames = function (nameMutex) {
+  if (!nameMutex.count) {
+    nameMutex.count = 1
+  }
+  this._uniqueName = 'C' + nameMutex.count++
+}
+
+/** @return {Object} */
+ConditionBuilder.prototype.buildAttributeNames = function () {
+  return typeUtil.buildAttributeNames(Object.keys(this._conditions))
+}
+
+/** @return {Object} */
+ConditionBuilder.prototype.buildAttributeValues = function () {
+  var result = {}
+  Object.keys(this._conditions).map(function (key) {
+    var list = this._conditions[key].AttributeValueList
+    if (!list) return
+
+    list.map(function (value, index) {
+      result[this._getValueAlias(key, index)] = value
+    }, this)
+  }, this)
+  return result
+}
+
+/**
+ * @see http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ExpressionPlaceholders.html#ExpressionAttributeValues
+ */
+ConditionBuilder.prototype._getValueAlias = function (key, index) {
+  if (!this._uniqueName) throw new Error('Names have not been assigned yet')
+  return ':V' + this._uniqueName + 'X' + key + 'X' + index
+}
+
+/**
+ * @return {string} String suitable for FilterExpression and KeyExpression
```

Comment:
mind to add a `@see` link to http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.SpecifyingConditions.html#ConditionExpressionReference.Syntax?